src/api/cpp/array.cpp

```diff
@@ -471,21 +471,31 @@ array::array_proxy &af::array::array_proxy::operator=(const array &other) {
         }
     }
 
-    af_array par_arr = nullptr;
+    af_array par_arr = 0;
 
+    dim4 dims = impl->parent_->dims();
```

Comment:
Can you name this parent_dim or original_dim?

Reply:
Changed to `parent_dim`